Public/Src/Cache/ContentStore/Library/FileSystem/PassThroughFileSystem.cs

```diff
@@ -416,9 +416,9 @@ private static bool GetSequentialScanOnOpenStreamThresholdEnvVariable(out long r
 
         private async Task CopyFileInsideSemaphoreAsync(AbsolutePath sourcePath, AbsolutePath destinationPath, bool replaceExisting)
         {
-            // It is very important to call OpenInsideSemaphoreAsync and not to call OpenAsync method that will re-acquire the semaphore once again.
+            // It is very important to call OpenInternal and not to call OpenAsync method that will re-acquire the semaphore once again.
             // Violating this rule may cause a deadlock.
-            using (var readStream = await OpenInsideSemaphoreAsync(
+            using (var readStream = OpenInternal(

```

Comment:
>OpenInternal [](start = 36, length = 12)

OpenInsideSemaphoreInternal?

Reply:
I got rid of the "InsideSemphore" part on purpose, because to me that implies that the method provides the semaphore, not that the method is intended to be called from another method that provides the semaphore.